bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/JsonCodecV2.java

```diff
@@ -0,0 +1,66 @@
+/********************************************************************************
+ * Copyright (c) 2022 STMicroelectronics.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.emfcloud.modelserver.emf.common.codecs;
+
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emfcloud.modelserver.common.ModelServerPathsV2;
+import org.eclipse.emfcloud.modelserver.common.codecs.DefaultJsonCodec;
+import org.eclipse.emfcloud.modelserver.common.codecs.EMFJsonConverter;
+import org.eclipse.emfcloud.modelserver.common.codecs.EncodingException;
+import org.eclipse.emfcloud.modelserver.emf.common.codecs.jackson.EMFModuleV2;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+/**
+ * A {@link JsonCodec}, adapted to support {@link ModelServerPathsV2} API V2 results.
+ */
+public class JsonCodecV2 extends DefaultJsonCodec {
+
+   public JsonCodecV2() {
+      super(createEMFJsonConverterV2());
+   }
+
+   private static EMFJsonConverter createEMFJsonConverterV2() {
+      EMFJsonConverter emfJsonConverter = new EMFJsonConverter();
+      emfJsonConverter.setMapper(setupObjectMapperV2());
+      return emfJsonConverter;
+   }
+
+   private static ObjectMapper setupObjectMapperV2() {
+      final ObjectMapper mapper = new ObjectMapper();
+      // same as emf
+      final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.ENGLISH);
+      dateFormat.setTimeZone(TimeZone.getDefault());
+
+      mapper.configure(SerializationFeature.INDENT_OUTPUT, true);
+      mapper.setDateFormat(dateFormat);
+      mapper.setTimeZone(TimeZone.getDefault());
+      mapper.registerModule(new EMFModuleV2());
+      return mapper;
+   }
+
+   @Override
+   public JsonNode encode(final EObject obj) throws EncodingException {
+      // FIXME: The super implementation creates a JsonResource, which doesn't (seem to) support
```

Comment:
In all custom tools, eg ecore-glsp or coffee-editor we are doing exactly the same and enable the ids. 
I'm wondering whether we should rather enable ids for the default JSONCodec and thus get rid of the JsonCodecV2 and the whole json2 resource identifier? @ndoschek @martin-fleck-at what do you think?

Reply:
Yes that sounds good to me and would simplify things already lot for any adopters.

Reply:
JsonV1 and JsonV2 have other differences, especially the way eClass and id are serialized. In V1, we have `{eClass: "EMF Type"}`, whereas in V2 we have `{$type: "EMF Type", $id: "EMF Fragment"}

So even if we improve the support for IDs in V1, we still need to distinguish the codecs.

Reply:
Could this be tracked in a task somewhere and referenced here so we do not forget about it?